src/api/stats/arbitrum/getArbidexApys.ts

```diff
@@ -0,0 +1,145 @@
+const liquidityProviderFee = 0.0025;
+const tradingFeeClient = null;
+
+const getArbidexApys = async (): Promise<ApyBreakdownResult> => {
+  const tradingAprs = await getTradingAprs();
+  const farmApys = await getFarmApys();
+
+  return getApyBreakdown(pools, tradingAprs, farmApys, liquidityProviderFee);
+};
+
+const getTradingAprs = async () => {
+  let tradingAprs = {};
+
+  if (tradingFeeClient && liquidityProviderFee) {
+    const pairAddresses = pools.map(pool => pool.address.toLowerCase());
+    tradingAprs = await getTradingFeeApr(tradingFeeClient, pairAddresses, liquidityProviderFee);
+  }
+  return tradingAprs;
+};
+
+const getFarmApys = async (): Promise<BigNumber[]> => {
+  const apys: BigNumber[] = [];
+
+  const arxPrice = await fetchPrice({ oracle: oracle, id: oracleId });
+  const wethPrice = await fetchPrice({ oracle: oracle, id: nativeId });
+
+  const { arxRewardsPerSec, WETHRewardsPerSec, arxTotalAllocPoint, WETHTotalAllocPoint } =
+    await getMasterChefData();
+  const { balances, arxAllocPoints, WETHAllocPoints } = await getPoolsData();
+
+  if (log) {
+    console.log(
+      'arxPerSec',
+      arxRewardsPerSec.div(decimals).toNumber(),
+      'WETHPerSec',
+      WETHRewardsPerSec.div(nativeDecimals).toNumber(),
+      'arxTotalAllocPoint',
+      arxTotalAllocPoint.toNumber(),
+      'WETHTotalAllocPoint',
+      WETHTotalAllocPoint
+    );
+  }
+
+  for (let i = 0; i < pools.length; i++) {
+    const pool = pools[i];
+
+    const oracle = pool.oracle ?? 'lps';
+    const id = pool.oracleId ?? pool.name;
+    const stakedPrice = await fetchPrice({ oracle, id });
+    const totalStakedInUsd = balances[i].times(stakedPrice).dividedBy(pool.decimals ?? '1e18');
+
+    const poolArxRewards = arxRewardsPerSec
+      .times(arxAllocPoints[i])
+      .dividedBy(arxTotalAllocPoint)
+      .times(1 - (pool.depositFee ?? 0));
+
+    const poolWethRewards = WETHRewardsPerSec.times(WETHAllocPoints[i])
+      .dividedBy(WETHTotalAllocPoint)
+      .times(1 - (pool.depositFee ?? 0));
+
+    const secondsPerYear = 31536000;
+    const yearlyArxRewards = poolArxRewards.times(secondsPerYear);
+    const yearlyWethRewards = poolWethRewards.times(secondsPerYear);
+
+    const yearlyArxRewardsInUsd = yearlyArxRewards.times(arxPrice).dividedBy(decimals);
+    const yearlyWethRewardsInUsd = yearlyWethRewards.times(wethPrice).dividedBy(nativeDecimals);
+
+    const yearlyRewardsInUsd = yearlyArxRewardsInUsd.plus(yearlyWethRewardsInUsd);
+
+    const apy = yearlyRewardsInUsd.dividedBy(totalStakedInUsd);
+    apys.push(apy);
+    if (log) {
+      console.log(
+        pool.name,
+        apy.toNumber(),
+        totalStakedInUsd.valueOf(),
+        yearlyRewardsInUsd.valueOf()
+      );
+    }
+  }
+
+  return apys;
+};
+
+const getMasterChefData = async () => {
+  const masterchefContract = getContractWithProvider(abi, masterchef, web3);
+
+  const arxRewardsPerSec = new BigNumber(await masterchefContract.methods.arxPerSec().call());
```

Comment:
this could be 1 multicall instead of 4 calls.
or ideally just combine it with existing multicall in getPoolsData
```js
const chefCalls = [{
  arxPerSec: masterchefContract.methods.arxPerSec(),
  WETHPerSec: masterchefContract.methods.WETHPerSec(),
  arxTotalAllocPoint: masterchefContract.methods.arxTotalAllocPoint(),
  WETHTotalAllocPoint: masterchefContract.methods.WETHTotalAllocPoint(),
}];

const res = await multicall.all([balanceCalls, allocPointCalls, chefCalls]);
const chefData = res[2].map(v => ({
  arxPerSec: new BigNumber(v.arxPerSec),
  // other fields
}))[0];
```

Reply:
@roman-monk I combined the chef calls and pool calls into a single multicall as you suggested.